src/v/cluster/data_policy_manager.cc

```diff
@@ -25,27 +25,25 @@ namespace cluster {
 namespace {
 
 template<typename Cmd>
-std::error_code do_apply(Cmd cmd, v8_engine::data_policy_table& db) {
+ss::future<std::error_code> do_apply(Cmd cmd, v8_engine::api& v8_api) {
     return ss::visit(
       std::move(cmd),
-      [&db](create_data_policy_cmd cmd) {
-          return db.insert(cmd.key, std::move(cmd.value.dp))
-                   ? std::error_code(errc::success)
-                   : std::error_code(errc::data_policy_already_exists);
+      [&v8_api](create_data_policy_cmd cmd) -> ss::future<std::error_code> {
+          return v8_api.insert(cmd.key, std::move(cmd.value.dp));
       },
-      [&db](delete_data_policy_cmd cmd) {
-          return db.erase(cmd.key)
-                   ? std::error_code(errc::success)
-                   : std::error_code(errc::data_policy_not_exists);
+      [&v8_api](delete_data_policy_cmd cmd) -> ss::future<std::error_code> {
+          co_return v8_api.remove(cmd.key);
       });
 }
 
 template<typename Cmd>
-ss::future<std::error_code> dispatch_updates_to_cores(
-  Cmd cmd, ss::sharded<v8_engine::data_policy_table>& db) {
-    auto res = co_await db.map_reduce0(
-      [cmd](v8_engine::data_policy_table& local_db) {
-          return do_apply(std::move(cmd), local_db);
+ss::future<std::error_code>
+dispatch_updates_to_cores(Cmd cmd, v8_engine::api& v8_api) {
+    auto res = co_await ss::map_reduce(
```

Comment:
`v8_api.map_reduce()` if v8_api is a sharded service and you intend to make a call on all shards, maybe this class should take a reference to `ss:;sharded<v8_api>` instead of `v8_api`

Reply:
I agree with rob. avoid raw `ss::map_reduce` and `ss::amp::submit_to` in favor of the sharded<>::xyz equivalents when possible.

Reply:
Is it ok, if sharded service stores sharded_services inside?
Of course I will construct v8_api only on 1 core.

Reply:
O think it is not good idea to do api - sharded
It contains sharded service inside and if I'll do it sharded I need to store this sharded service in another place...

Reply:
we have sharded services that store references to sharded service in many portions of the codebase. The main reason is because it might be desired for the current shard to perform work on the other sharded service on a different shard.

Reply:
> we have sharded services that store references to sharded service in many portions of the codebase. The main reason is because it might be desired for the current shard to perform work on the other sharded service on a different shard. So in my opinion you should do this because it is consistent with common patterns we use elsewhere too.

I understand way to store ref for another sharded_service
But v8_api is container and place for storing all v8 stuff
If I want to make it sharded I need to move another sharded service in another place. But we do not wnat to store v8 services in application

As a result, I need api which will store ref to sharded services and container where they will be located